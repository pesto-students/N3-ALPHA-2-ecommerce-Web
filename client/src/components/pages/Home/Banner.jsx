import React, { Fragment, useState } from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import SkeltonLoader from '../../shared/Loaders/SkeltonLoader';

const Banner = () => {
    const [showImgLoader, setImgLoader] = useState(true);
    const { t } = useTranslation();
    if (showImgLoader) {
        const objImg = new Image();
        objImg.src = '/assets/banner.jpg';
        objImg.onload = () => {
            setImgLoader(false);
        };
    }
    return (
        <section className="banner">
            {showImgLoader ? (
                <SkeltonLoader width={'100%'} height={'400px'} />
            ) : (
                <Fragment>
                    <img
                        src="/assets/banner.jpg"
                        className="banner_img"
                        alt="Banner"
                    />
                    <div className="banner_content">
                        <h1 className="banner_head">{t('header_heading')}</h1>
                        <p className="banner_text">{t('header_text')}</p>
                        <Link
                            to="/products?category=all"
                            className="banner_btn"
                        >
                            {t('shop_text')}
                        </Link>
                    </div>
                </Fragment>
            )}
        </section>
    );
};

export default Banner;
