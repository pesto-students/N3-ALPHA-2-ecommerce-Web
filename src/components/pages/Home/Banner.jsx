import React from 'react';
import { Link } from 'react-router-dom';

const Banner = () => {
    return (
        <section className="banner">
            <img src="/assets/banner.jpg" className="banner_img" alt="Banner" />
            <div className="banner_content">
                <h1 className="banner_head">Prevention is better than cure</h1>
                <p className="banner_text">
                    Order now and get things delivered within 48hrs.
                </p>
                <Link to="/products?category=all" className="banner_btn">
                    Shop Now
                </Link>
            </div>
        </section>
    );
};

export default Banner;
