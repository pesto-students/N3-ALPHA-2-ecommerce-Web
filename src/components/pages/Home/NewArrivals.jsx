import React from 'react';
import ProductItem from '../../shared/ProductItem/ProdcutItem';

const NewArrivals = (props) => {
    const { heading = '', selectedProd = [] } = props;
    return (
        <section className="newArrivals">
            <h1 className="newArrivals_heading">{heading}</h1>
            <div className="newArrivals_content">
                {selectedProd.slice(0, 4).map((itemData, index) => {
                    const {
                        thumbnail = '',
                        name = '',
                        price = '',
                        id = '',
                    } = itemData;
                    return (
                        <ProductItem
                            name={name}
                            img={`/assets/${thumbnail}`}
                            price={price}
                            id={id}
                        />
                    );
                })}
            </div>
        </section>
    );
};

export default NewArrivals;
