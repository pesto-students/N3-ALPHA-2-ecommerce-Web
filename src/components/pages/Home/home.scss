@import '../../shared/Style/index';

// Banner
.banner {
    position: relative;
    max-width: 1600px;
    margin: auto;
    &_img {
        width: 100%;
        max-width: 1600px;
        margin: auto;
        display: block;
        height: 400px;
        object-fit: cover;
    }
    &_content {
        position: absolute;
        top: 0;
        left: 10%;
        height: 100%;
        width: 30%;
        display: flex;
        flex-direction: column;
        justify-content: center;
        color: $white;
        text-align: left;
    }

    &_head {
        font-size: 3rem;
        margin: 0;
        color: $white;
        line-height: 1.2;
    }

    &_text {
        font-size: 1.8rem;
    }

    &_btn {
        @include hyGenieBtn;
        width: 28%;
        text-align: center;
        text-decoration: none;
    }
}

// New Arrivals
.newArrivals {
    @include sectionPadding;
    &_heading {
        @include sectionHeading;
    }
    &_content {
        @include sectionWidth;
        display: flex;
        .product-item {
            width: 25%;
        }
    }
    &_list {
        @include productCard;
        &-img {
            max-height: 200px;
        }
    }
}

// desktop and ipad pro
@media only screen and (min-width: $mediaLaptopMin) and (max-width: $mediaLaptopMax) {
    .banner {
        &_img {
            height: 320px;
        }
        &_content {
            width: 40%;
            left: 4%;
        }
    }
}

// for  tablets
@media (max-width: $mediaTabMax) {
    .banner {
        &_img {
            height: 260px;
        }
        &_content {
            width: 40%;
            left: 4%;
        }
        &_head {
            font-size: 2rem;
        }
        &_text {
            font-size: 1.4rem;
        }
        &_btn {
            width: 32%;
        }
    }
    .newArrivals {
        &_content {
            flex-wrap: wrap;
            .product-item {
                width: 48.8%;
                margin-right: $margin;
                &:nth-child(even) {
                    margin-right: 0;
                }
            }
        }
    }
}

@media only screen and (min-width: 481px) and (max-width: 720px) {
    .newArrivals {
        &_content {
            .product-item {
                width: 48%;
            }
        }
    }
}

// for mobile device
@media (max-width: $mediaTabMin) {
    .banner {
        &_img {
            display: none;
        }
        &_content {
            width: 100%;
            position: static;
            background: $black-343A40;
            padding: 16px;
            box-sizing: border-box;
        }
        &_head {
            font-size: 2rem;
        }
        &_text {
            font-size: 1.4rem;
        }
        &_btn {
            width: auto;
        }
    }
    .newArrivals {
        &_content {
            flex-wrap: wrap;
            .product-item {
                width: 100%;
                margin-right: 0;
            }
        }
    }
}
