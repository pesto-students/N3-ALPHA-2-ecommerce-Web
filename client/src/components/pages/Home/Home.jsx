import React, { Fragment, useEffect } from 'react';
import Banner from './Banner';
import NewArrivals from './NewArrivals';
import { getSessionStorage, toaster } from '../../../helper/Utils';
import FullPageLoader from '../../shared/Loaders/FullPageLoader';
import { useTranslation } from 'react-i18next';

const Home = () => {
    const newArrivals = getSessionStorage('arrivals') || [];
    const recomended = getSessionStorage('recomended') || [];
    const { t } = useTranslation();
    useEffect(() => {
        document.title = 'HyGenie | Stay Home Stay Safe';
    }, []);
    return (
        <Fragment>
            {recomended.length > 0 ? (
                <Fragment>
                    <Banner />
                    <NewArrivals
                        heading={t('newArrivals_text')}
                        selectedProd={newArrivals}
                    />
                    <NewArrivals
                        heading={t('recomended_text')}
                        selectedProd={recomended}
                    />
                </Fragment>
            ) : (
                <FullPageLoader />
            )}
        </Fragment>
    );
};

export default Home;
