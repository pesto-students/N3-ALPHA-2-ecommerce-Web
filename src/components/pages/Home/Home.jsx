import React, { Fragment } from 'react';
import Banner from './Banner';
import NewArrivals from './NewArrivals';
import { getSessionStorage } from '../../../helper/Utils';
import FullPageLoader from '../../shared/Loaders/FullPageLoader';

const Home = () => {
    const newArrivals = getSessionStorage('arrivals') || [];
    const recomended = getSessionStorage('recomended') || [];
    return (
        <Fragment>
            {recomended.length > 0 ? (
                <Fragment>
                    <Banner />
                    <NewArrivals
                        heading="New Arrivals"
                        selectedProd={newArrivals}
                    />
                    <NewArrivals
                        heading="Recommended"
                        selectedProd={recomended}
                    />
                </Fragment>
            ) : (
                <FullPageLoader />
            )}
        </Fragment>
    );
};

export default Home;
