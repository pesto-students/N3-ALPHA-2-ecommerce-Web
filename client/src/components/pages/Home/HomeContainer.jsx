import React, { Fragment, useEffect } from 'react';
import Home from './Home';
import './home.scss';
import useGetAllProducts from '../../shared/Hooks/useGetAllProducts';
import FullPageLoader from '../../shared/Loaders/FullPageLoader';

const HomeContainer = () => {
    const allProducts = useGetAllProducts();
    useEffect(() => {}, [allProducts]);

    return (
        <Fragment>
            {allProducts.length > 0 ? <Home /> : <FullPageLoader />}
        </Fragment>
    );
};

export default HomeContainer;
