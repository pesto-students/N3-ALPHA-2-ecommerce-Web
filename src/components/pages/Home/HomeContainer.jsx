import React, { Fragment } from 'react';
import HomeComponent from './Home';
import './home.scss';
import useGetAllProducts from '../../shared/Hooks/useGetAllProducts';
import { getRandomArrayItem, sliceArray } from '../../../helper/Utils';
import useSessionStorage from '../../shared/Hooks/useSessionStorage';

const HomeContainer = () => {
    const allProducts = useGetAllProducts(),
        selectedProd = getRandomArrayItem(allProducts, 8),
        newArrivals = sliceArray(selectedProd, 0, 4),
        recomended = sliceArray(selectedProd, 4, 8);
    const [arrivals] = useSessionStorage('arrivals', newArrivals);
    const [allRecomended] = useSessionStorage('recomended', recomended);
    return (
        <Fragment>
            <HomeComponent newArrivals={arrivals} recomended={allRecomended} />
        </Fragment>
    );
};

export default HomeContainer;
