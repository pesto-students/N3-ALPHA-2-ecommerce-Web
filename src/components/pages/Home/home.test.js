import renderer from 'react-test-renderer';
import Home from './Home';

test('Home Snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
});
