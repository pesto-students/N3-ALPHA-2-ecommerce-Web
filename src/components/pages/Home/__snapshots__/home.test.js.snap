// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot 1`] = `
<div
  className="fullPageWrap"
>
  <div
    className="fullPageLoader"
  />
</div>
`;
